students/marsha_w/lesson_06/test_mailroom.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+
+'''Title: Unit testing for create a report function
+Dev: Marsha Wheeler
+Date: 02/23/2019
+'''
+
+from mailroom4 import create_report, write_text, write_file, send_thank_you, donor_dict
+import os.path
+import datetime
+
+# test create report function
+def test_donationsum_order():
+    report_dict = create_report()
+    assert report_dict[0][1] > report_dict[1][1]
+    assert report_dict[0][1] > report_dict[2][1]
+    assert report_dict[1][1] > report_dict[2][1]
+
+
+def test_num_donations(donor="William Gates"):
+    report_dict = create_report()
+    dict_num = len(donor_dict[donor])
+    for i in report_dict:
+        if donor in i:
+            assert dict_num == (i[2])
+
+def test_check_sum(donor="Mark Zuckerberg"):
+    report_dict = create_report()
+    dict_sum = sum(donor_dict[donor])
+    assert dict_sum == report_dict[1][1]
+
+
+def test_check_mean(donor="Paul Allen"):
+    report_dict = create_report()
+    dict_mean = sum(donor_dict[donor]) / len(donor_dict[donor])
+    assert dict_mean == report_dict[3][3]
+
+#test write email functions
+
+def test_write_text():
+    expected_text = "Dear Jeff Bezos, \n We are writing to thank you for your generous total donation of $877.33 " \
+                 "to our organization. \n Sincerely, "
+    assert expected_text == write_text("Jeff Bezos")
+
+
+def test_write_text2():
+    expected_text = "Dear William Gates, \n We are writing to thank you for your generous total donation of $653784.49 " \
+                 "to our organization. \n Sincerely, "
+    assert expected_text == write_text("William Gates")
+
+
+def test_write_file():
+    write_file()
+    for keys in donor_dict:
+        date = str(datetime.datetime.now()).split()
+        out_name = str(keys.replace(' ', '_') + '_' + date[0] + ".txt")
+        assert os.path.isfile(
+            '/Users/marshawheeler/GP_Python210B_Winter_2019/students/marsha_w/lesson_06/' + out_name)
+
+
+#test appending send_thank_you function
+
+def test_donor_name1(donor="Paul Allen", amount=float(500)):
```

Comment:
not sure why you included donor, amount as function variables here.  What would have been interesting is if you abstracted this idea out for testing.  For example:
```
def _validate_thank_you(donor, amount):
    send_thank_you(donor, amount)
    assert donor in donor_dict
```
And have your tests call that function passing in the donor and mount.  Keep in mind though, you only need to test it once rather than with multiple donors and amounts each time, as you don't really gain anything by testing the same functionality 4-5 times if that makes sense from a programming perspective, unless those inputs where slightly different each time.